server/schedulers/evict_slow_trend.go

```diff
@@ -0,0 +1,437 @@
+	return conf.evictCandidate
+}
+
+func (conf *evictSlowTrendSchedulerConfig) candidateCapturedSecs() uint64 {
+	return uint64(time.Now().Sub(conf.candidateCaptureTime).Seconds())
+}
+
+func (conf *evictSlowTrendSchedulerConfig) captureCandidate(id uint64) {
+	conf.evictCandidate = id
+	conf.candidateCaptureTime = time.Now()
+}
+
+func (conf *evictSlowTrendSchedulerConfig) popCandidate() uint64 {
+	id := conf.evictCandidate
+	conf.evictCandidate = 0
+	return id
+}
+
+func (conf *evictSlowTrendSchedulerConfig) setStoreAndPersist(id uint64) error {
+	conf.EvictedStores = []uint64{id}
+	return conf.Persist()
+}
+
+func (conf *evictSlowTrendSchedulerConfig) clearAndPersist(cluster schedule.Cluster) (oldID uint64, err error) {
+	oldID = conf.evictedStore()
+	if oldID == 0 {
+		return
+	}
+	address := "?"
+	store := cluster.GetStore(oldID)
+	if store != nil {
+		address = store.GetAddress()
+	}
+	storeSlowTrendEvictedStatusGauge.WithLabelValues(address, strconv.FormatUint(oldID, 10)).Set(0)
+	conf.EvictedStores = []uint64{}
+	return oldID, conf.Persist()
+}
+
+type evictSlowTrendScheduler struct {
+	*BaseScheduler
+	conf *evictSlowTrendSchedulerConfig
+}
+
+func (s *evictSlowTrendScheduler) GetName() string {
+	return EvictSlowTrendName
+}
+
+func (s *evictSlowTrendScheduler) GetType() string {
+	return EvictSlowTrendType
+}
+
+func (s *evictSlowTrendScheduler) EncodeConfig() ([]byte, error) {
+	return schedule.EncodeConfig(s.conf)
+}
+
+func (s *evictSlowTrendScheduler) Prepare(cluster schedule.Cluster) error {
+	evictedStoreID := s.conf.evictedStore()
+	if evictedStoreID == 0 {
+		return nil
+	}
+	return cluster.SlowTrendEvicted(evictedStoreID)
+}
+
+func (s *evictSlowTrendScheduler) Cleanup(cluster schedule.Cluster) {
+	s.cleanupEvictLeader(cluster)
+}
+
+func (s *evictSlowTrendScheduler) prepareEvictLeader(cluster schedule.Cluster, storeID uint64) error {
+	err := s.conf.setStoreAndPersist(storeID)
+	if err != nil {
+		log.Info("evict-slow-trend-scheduler persist config failed", zap.Uint64("store-id", storeID))
+		return err
+	}
+	return cluster.SlowTrendEvicted(storeID)
+}
+
+func (s *evictSlowTrendScheduler) cleanupEvictLeader(cluster schedule.Cluster) {
+	evictedStoreID, err := s.conf.clearAndPersist(cluster)
+	if err != nil {
+		log.Info("evict-slow-trend-scheduler persist config failed", zap.Uint64("store-id", evictedStoreID))
+	}
+	if evictedStoreID != 0 {
+		cluster.SlowTrendRecovered(evictedStoreID)
+	}
+}
+
+func (s *evictSlowTrendScheduler) scheduleEvictLeader(cluster schedule.Cluster) []*operator.Operator {
+	store := cluster.GetStore(s.conf.evictedStore())
+	if store == nil {
+		return nil
+	}
+	storeSlowTrendEvictedStatusGauge.WithLabelValues(store.GetAddress(), strconv.FormatUint(store.GetID(), 10)).Set(1)
+	return scheduleEvictLeaderBatch(s.GetName(), s.GetType(), cluster, s.conf, EvictLeaderBatchSize)
+}
+
+func (s *evictSlowTrendScheduler) IsScheduleAllowed(cluster schedule.Cluster) bool {
```

Comment:
Defined but not used, could we test it in related uts?

Reply:
It's copied from other schedulers, I don't think it need an extra test